src/ultk/language/language.py

```diff
@@ -15,19 +15,20 @@
 import numpy as np
 from ultk.language.semantics import Universe
 from ultk.language.semantics import Meaning, Referent
+from dataclasses import dataclass
 from typing import Callable, Iterable
 
 
+@dataclass()
 class Expression:
 
     """Minimally contains a form and a meaning."""
 
-    def __init__(self, form: str | None = None, meaning: Meaning | None = None):
-        # gneric/dummy form and meaning if not specified
-        # useful for hashing in certain cases
-        # (e.g. a GrammaticalExpression which has not yet been evaluate()'d and so does not yet have a Meaning)
-        self.form = form or ""
-        self.meaning = meaning or Meaning(tuple([]), Universe(tuple([])))
+    # gneric/dummy form and meaning if not specified
+    # useful for hashing in certain cases
+    # (e.g. a GrammaticalExpression which has not yet been evaluate()'d and so does not yet have a Meaning)
+    form: str | None = ""
+    meaning: Meaning | None = Meaning(tuple(list()), Universe(tuple(list())))
```

Comment:
So, the empty string and the "dummy" `Meaning` on the rhs of these assignments are default values.  If we use them, I don't think we need the `| None` at all.  Or we can make `None` be the default.  I think I lean towards the former; what do you think?

Reply:
Indeed, best to not allow None!